File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {

        int n = cardPoints.size();
        int i = 0, j = 0;
        int total = 0;
        int ans = 0;

        while(j<k) {
            total+=cardPoints[j];
            j++;
        }

        j= n-1;
        i = k-1;
        ans = total;

        while(i>=0) {
            total-=cardPoints[i];
            i--;
            total+=cardPoints[j];
            j--;
            ans = max(ans,total);
        }

        return ans;
    }
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] arr, int k) {
        int n = arr.length;

        int i = n-k;
        int ans = 0;
        int sum = 0;
        int j = i;
        boolean flag = true;

        while(i<n) {
            
            while(j<n && flag && j-i+1<=k) {
                sum+=arr[j];
                
                if(j-i+1==k) {
                    ans = Math.max(sum,ans);
                    //System.out.println("inner "+ans);
                    flag = false;
                }
                j++;

            }
            
            j = j%n;
            

            sum-=arr[i];
            i++;
            sum+=arr[j];
            ans = Math.max(ans,sum);
            j++;
           
           
        }

        return ans;
    }
}